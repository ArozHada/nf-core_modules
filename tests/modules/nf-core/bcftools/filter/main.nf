#!/usr/bin/env nextflow

nextflow.enable.dsl = 2
moduleDir = launchDir

//keep --no-verson argument, otherwise md5 will change on each execution
include { BCFTOOLS_FILTER } from "$moduleDir/modules/nf-core/bcftools/filter/main.nf"

workflow test_bcftools_filter {
    input = [ [ id:'test' ], // meta map
              [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
            ]

    BCFTOOLS_FILTER ( input )
}
