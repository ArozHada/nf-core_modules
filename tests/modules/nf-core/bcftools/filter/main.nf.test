nextflow_process {

    name "Test Process BCFTOOLS_FILTER"
    script "modules/nf-core/bcftools/filter/main.nf"
    process "BCFTOOLS_FILTER"
    config "tests/modules/nf-core/bcftools/filter/nextflow.config"
    tag "bcftools_filter"

    test("Sarscov2 Illumina VCF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                             [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
