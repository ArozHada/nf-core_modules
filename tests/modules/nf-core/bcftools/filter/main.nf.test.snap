{
    "Sarscov2 Illumina VCF": {
        "content": [
            [
                "test.vcf.gz:md5,4f24467157f5c7a3b336481acf0c8a65",
                "versions.yml:md5,e6c13c4a00c70667184f57c65ebfdeb8"
            ]
        ],
        "timestamp": "2023-04-09T14:03:09+0000"
    }
}